### DISCUSSION TITLE
Animated PNG to WebP?

### DISCUSSION THREAD

**Discussion by SoftCreatR** [ImageMagick/ImageMagick#4678] [Help] [answered]
I'm currently trying to convert animated PNGs to WebP, but all I get are static WebP files, when running

```bash
for file in *.png; do convert $file "`basename $file .png`.webp"; done
```

Example file:

![butterfly@2x](https://user-images.githubusercontent.com/81188/148636196-58e701f1-983a-4ec5-856e-f55c0f31c2f8.png)

Version I use is  ImageMagick 7.1.0-19 Q16.

**Comment by SoftCreatR** [comment]
Update: I found [this little Python script](https://github.com/Benny-/apng2webp), that works fine, so my problem is solved for now. But it would be nice, if I can make it work using IM too, because that Python script is pretty old, doesn't support quality settings, etc.

**Comment by fmw42** [comment, marked as answer]
You have an animated PNG so you need to tell IM that.

`magick -background white -dispose background APNG:butterfly.png butterfly.webp`

however, there seems to be a dispose issue that I cannot solve. **So I will leave this to the IM developers.**

I also tried:

`magick -background white -dispose background APNG:butterfly.png -coalesce butterfly.webp`

Be sure you have the current Webp delegate library installed.

See https://imagemagick.org/script/webp.php

**Reply by SoftCreatR** [reply]
Ty, that seems to work. But as you've already pointed out, there seems to be an issue, so the resulting WebP is somewhat "slow", compared to the original.
